tests/rptest/tests/tiered_storage_model_test.py

```diff
@@ -0,0 +1,551 @@
+                validator.run(self)
+            except:
+                self.logger.error("Failed to run validator", exc_info=True)
+
+        self._bg_loop(invoke, validator.name(),
+                      lambda: random.uniform(0.5, 1.5))
+
+    def run_bg_validators(self, test_case):
+        """Run all validators in background."""
+        # Start all bg-loops which are needed to run validators
+        self._bg_grep_logs()
+        self._bg_metrics_pull(MetricsEndpoint.PUBLIC_METRICS)
+        self._bg_metrics_pull(MetricsEndpoint.METRICS)
+        self._bg_bucket_view_update()
+        # Start validators running in the background
+        for v in test_case.validators():
+            self._bg_validator_run(v)
+
+    def _bg_metrics_pull(self, endpoint):
+        """Pull metrics from the redpanda node. This is needed to avoid pulling metrics
+        from every validator individually. The validators are invoking functions of the
+        test suite to find individual metrics."""
+        try:
+            if self.stop_flag:
+                return
+            res = defaultdict(int)
+            for n in self.redpanda.nodes:
+                if n in self.redpanda._started:
+                    resp = self.redpanda.metrics(n, endpoint)
+                    for family in resp:
+                        for sample in family.samples:
+                            res[sample.name] += sample.value
+            # should be safe due to GIL
+            if endpoint == MetricsEndpoint.PUBLIC_METRICS:
+                self.public_metrics = res
+                self.logger.info(f"Public metrics {res}")
+            else:
+                self.private_metrics = res
+                self.logger.info(f"Private metrics {res}")
+            time.sleep(2)
+        except:
+            self.logger.error("Failed to pull metrics", exc_info=True)
+            time.sleep(2)
+        finally:
+            # Run 'loop' recursively
+            if not self.stop_flag:
+                self.thread_pool.submit(self._bg_metrics_pull, endpoint)
+
+    def _bg_bucket_view_update(self):
+        """Re-scan the bucket view in the background."""
+        try:
+            if self.stop_flag:
+                return
+            self.bucket_view = BucketView(self.redpanda,
+                                          topics=self.topics,
+                                          scan_segments=True)
+            self.bucket_view._ensure_listing()
+            self.logger.info(
+                f"Bucket view updated, {self.bucket_view.segment_objects} segments scanned"
+            )
+            time.sleep(10)
+        except:
+            self.logger.error("Failed to scan bucket", exc_info=True)
+            time.sleep(20)
+        finally:
+            if not self.stop_flag:
+                self.thread_pool.submit(self._bg_bucket_view_update)
+
+    def _bg_grep_logs(self):
+        if self.stop_flag:
+            return
+
+        def tail_minus_f(node):
```

Comment:
Is it possible that we miss some logs between the invocations of `_bg_grep_logs`? If so, is that a problem?

Reply:
we shouldn't miss anything because the tail -f is called at the beginning, before any validator is even started and before the test is doing anything